pcsx2/Recording/InputRecordingFile.cpp

```diff
@@ -36,7 +37,12 @@ void InputRecordingFile::InputRecordingFileHeader::init() noexcept
 
 void InputRecordingFile::setEmulatorVersion()
 {
-	static const std::string emuVersion = fmt::format("PCSX2-{}.{}.{}", PCSX2_VersionHi, PCSX2_VersionMid, PCSX2_VersionLo);
+	std::string emuVersion;
+	if (!PCSX2_isReleaseVersion && GIT_TAGGED_COMMIT)
+		emuVersion = fmt::format("PCSX2-Nightly-{}", GIT_TAG);
+	else
+		emuVersion = fmt::format("PCSX2-{}.{}.{}", PCSX2_VersionHi, PCSX2_VersionMid, PCSX2_VersionLo);
+
 	strncpy(m_header.m_emulatorVersion, emuVersion.c_str(), sizeof(m_header.m_emulatorVersion) - 1);
```

Comment:
`strncpy()` doesn't guarantee null termination. Please use `StringUtil::Strlcpy()` instead - can get rid of the silly -1 at the end too then.

Should probably zero out the rest of the bytes too - add `= {}` to the header here <[#10144](https://github.com/PCSX2/pcsx2/blob/master/pcsx2/Recording/InputRecordingFile.h#L36)>